src/rake.rs

```diff
@@ -26,7 +26,16 @@ impl Rake {
     /// Runs RAKE algorithm on `text` and returns a vector of keywords.
     /// The returned vector is sorted by score (from greater to less).
     pub fn run(&self, text: &str) -> Vec<KeywordScore> {
-        let phrases = self.phrases(text);
+        let phrases = self.phrases(PUNC_RE.split(text));
+        let word_scores = self.word_scores(&phrases);
+        self.candidate_keywords(&phrases, word_scores)
+    }
+
+    /// Runs RAKE algorithm on chunks of text and returns a vector of keywords.
+    /// The returned vector is sorted by score (from greater to less).
+    #[inline]
+    pub fn run_sentences<'a>(&self, sentences: impl IntoIterator<Item = &'a str>) -> Vec<KeywordScore> {
```

Comment:
1. I think it should be renamed to `run_by_iterator` or something like that.
2. this function might return wrong result by passing an invalid iterator. So, it should be mentioned in the documentation of this function.

Reply:
I think naming it after a language construct is not correct. It should use domain-specific name. Same as you use `let phrases =`, not `let variable = `.

Reply:
I agree, but `run_sentences` is somewhat ambiguous.

Reply:
phrases? chunks? fragments? I'm not sure what's the technically best terminology here. The implementation used to split on punctuation which is roughly equivalent to sentences?